source/iostreams.tex

```diff
@@ -15751,8 +15753,14 @@
 
 \begin{itemdescr}
 \pnum
-\returns An unspecified directory path suitable for temporary files. An error shall be reported if
-\tcode{exists(p)} is \tcode{false} or \tcode{is_directory(p)} is \tcode{false}, where \tcode{p} is the path to be returned.
+Let \tcode{p} be the path to be returned.
+
+\pnum
+\effects If \tcode{exists(p)} is \tcode{false} or \tcode{is_directory(p)} is
+  \tcode{false}, an error is reported\iref{fs.err.report}.
+
+\pnum
+\returns An unspecified directory path suitable for temporary files.
```

Comment:
Hm, I think it would be nice if we could make reference to `p` in the Returns element. Maybe the "unspecified directory path..." could move up into the defining paragraph, and we could say "returns `p`" here?

Reply:
Sounds like a good idea.